primitives/src/crh/bowe_hopwood/mod.rs

```diff
@@ -87,16 +87,17 @@ impl<G: Group, W: PedersenWindow, const N: usize> BoweHopwoodPedersenCRH<G, W, N
     }
 
     /// Zcash protocol specification - section 5.4.1.7
-    fn calculate_num_chunks_in_segment<F: PrimeField>() -> usize {
-        let upper_limit: BigUint = F::modulus_minus_one_div_two().into();
+    pub(crate) fn calculate_num_chunks_in_segment<F: PrimeField>() -> usize {
+        let upper_limit: BigUint = ((F::modulus_minus_one_div_two().into() * 15u32)/4u32) + 1u32;
+
         let mut c = 0;
         let mut range = BigUint::from(2usize);
```

Comment:
I realized now that this should probably be initialized to 1 rather than 2, as otherwise `range = 2^(4c+1) rather than 2^4c